cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -215,17 +260,19 @@ private void checkTetheringServerEnabled() throws NotImplementedException {
     }
   }
 
-  private void updateTetherStatus(HttpResponder responder, String peer, TetheringStatus newStatus)
-    throws NotImplementedException, PeerNotFoundException, IOException {
-    checkTetheringServerEnabled();
-    PeerInfo peerInfo = store.getPeer(peer);
-    if (peerInfo.getTetheringStatus() == TetheringStatus.PENDING) {
-      store.updatePeerStatus(peerInfo.getName(), newStatus);
-    } else {
-      LOG.info("Cannot update tether state to {} as current state state is {}",
-               newStatus, peerInfo.getTetheringStatus());
+  private void deleteTethering(String peer) throws IOException {
+    try {
+      store.deletePeer(peer);
+    } catch (PeerNotFoundException e) {
+      // Peer doesn't exist, nothing to do here
+    }
+    TopicId topic = new TopicId(NamespaceId.SYSTEM.getNamespace(),
+                                topicPrefix + peer);
+    try {
+      messagingService.deleteTopic(topic);
```

Comment:
what about when this fails? If somebody creates the peer again will it then have leftover messages, or will it be "empty" due to whatever message id the client sends over?

Reply:
It will have leftover messages.

Reply:
hm I'm not sure the best way to handle this, as we run into more problematic issues if we try to delete the topic before deleting the peer from the DB. 

One idea is to store the creation time of the peer, then for the very first poll, start consuming messages from that timestamp instead of the beginning of the topic. Not sure how feasible that is though. In any case, let's open a jira and add a TODO here.

Reply:
Added TODO